src/program.rs

```diff
@@ -54,6 +57,108 @@ impl std::str::FromStr for ColorOpt {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct QueryPath(pub Vec<Ident>);
+
+impl QueryPath {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn parse(cache: &mut Cache, input: String) -> Result<Self, ParseError> {
```

Comment:
This parser seems to be a special case of something we already do in the general Nickel parser. Namely, we need to understand field paths already for record definitions like
```
{ foo.bar.baz = ... }
```
It's probably not easy or reasonable to try to call into the generic parser from here. But is there some mechanism to keep these parsers in sync? For example, what if we start adding wildcard fields?

Reply:
In fact it's not too hard to call the parser: we just have to make the rule public. I just found it a bit too heavy then, but maybe it's a good idea. Would that be ok to let it that way for the time being, and to consider this option in a subsequent PR?

Reply:
I think leaving it the way it is would be perfectly reasonable. This is mostly a maintainability concern if we ever end up changing the accepted syntax for field paths. Which we shouldn't be doing often anyway.